### PR TITLE
[NuGetizer3000] Visual Studio on windows compatibility

### PR DESCRIPTION
**Pull request by mrward**
Fixes several NuGetizer 3000 bugs including:

 1. 56701 - Windows created nuspec can not be loaded on a mac
 2. 56706 - Can not create a nuget package with Standard net libraries
 3. 52657 - Creating a new Multiplatform Library requires a
description
 4. Packages folder empty on re-opening NuGet packaging project
 5. Fix Dependencies folder being displayed if latest NuGet.Build.Packaging 0.1.276 NuGet package is installed.
 6. Create MSBuild targets during restore for NuGet packaging project if they are missing.
 7. Update to latest stable NuGet.Build.Packaging 0.1.276 in project templates.

### CODE REVIEW

File: main/src/addins/MonoDevelop.DotNetCore/MonoDevelop.DotNetCore/DotNetCoreProjectExtension.cs [hunk 1]
```diff
@@ -74,9 +74,7 @@ protected override bool OnGetSupportsFramework (TargetFramework framework)
 
 		bool IsDotNetCoreProject (DotNetProject project)
 		{
-			var properties = project.MSBuildProject.EvaluatedProperties;
-			return properties.HasProperty ("TargetFramework") ||
-				properties.HasProperty ("TargetFrameworks");
+			return project.MSBuildProject.Sdk != null;
```
Comment (slluis) [inline main/src/addins/MonoDevelop.DotNetCore/MonoDevelop.DotNetCore/DotNetCoreProjectExtension.cs]: Is this really correct? A project may use an Sdk and not be a .net core project.

Comment (mrward) [inline main/src/addins/MonoDevelop.DotNetCore/MonoDevelop.DotNetCore/DotNetCoreProjectExtension.cs]: The method name is probably a bit misleading. There is another check made for the target framework elsewhere.

There is also another question of whether we should split out Sdk project support from the .NET Core project extension. Currently they are mixed together and probably should be separated out.

Comment (mrward) [inline main/src/addins/MonoDevelop.DotNetCore/MonoDevelop.DotNetCore/DotNetCoreProjectExtension.cs]: However it does look wrong so I will change it to check the target framework is supported.

**Comment by mrward** [comment]
Restricted the extension so it is only enabled for sdk projects that target .NET Core or .NET Standard.

**Comment by mrward** [comment]
I think restricting the supported frameworks to netcoreapp and netstandard will break the IoT addin since it currently uses net461 as its target framework. So for now I may have to allow .NETFramework.

File: main/src/addins/MonoDevelop.DotNetCore/MonoDevelop.DotNetCore/DotNetCoreProjectExtension.cs (hunk 1 above)
Comment (mhutch) [inline main/src/addins/MonoDevelop.DotNetCore/MonoDevelop.DotNetCore/DotNetCoreProjectExtension.cs]: +1 on separating them out.

**Comment by mhutch** [comment]
@mrward it'll break any non.NET Core project using sdks, surely? That includes Azure functions.

See also https://github.com/mono/monodevelop/commit/2b2c26b588755f1eb953555c2fc7cadb0c9b496e

**Comment by mrward** [comment]
Yeah I think I am going to revert aa14e80 for now. I think we should separate out sdk project file support from the DotNetCoreProjectExtension at some point.

**Comment by mrward** [comment]
Reverted the change restricting the extension.

**Comment by mhutch** [comment]
Agreed, and we should probably move it down to core seeing how Sdk projects are becoming more common.

**Pull request merged**